src/stdune/path.mli

```diff
@@ -1,30 +1,24 @@
-(** Relative path with unspecified root *)
+(** Relative path without restrictions *)
 module Relative : sig
   include Path_intf.S
 
-  (** [root] refers to empty relative path, so whatever the path is interpreted
-      relative to. *)
-  val root : t
-
   val of_string : string -> (t, string) Result.t
   val of_string_exn : string -> t
 
   module L : sig
-    val relative : t -> string list -> (t, string) Result.t
-    val relative_exn : t -> string list -> t
+    val relative : t -> string list -> t
   end
 
-  val relative : t -> string -> (t, string) Result.t
-  val relative_exn : t -> string -> t
-  val split_first_component : t -> (string * t) option
-  val explode : t -> string list
+  val relative : t -> string -> t
 end
 
-(** In the current workspace (anything under the current project root) *)
+(** relative path that isn't allowed to escape root *)
```

Comment:
Propose clarification:

(** Relative path that's not allowed to escape its base.
Its representation is normalized, so it's isomorphic to a list of components and it has no `.`, `..` or empty components.
The base of this path is unspecified (it is often something different than the source root). *)